qemu/tests/qemu_guest_agent.py

```diff
@@ -454,12 +454,44 @@ def gagent_check_set_user_password(self, test, params, env):
     @error_context.context_aware
     def gagent_check_get_vcpus(self, test, params, env):
         """
-        Execute "guest-set-vcpus" command to guest agent
+        Execute "guest-get-vcpus" command to guest agent.
+
+        Steps:
+        1) Check cpu info from guest agent.
+        2) Check cpu number.
+
         :param test: kvm test object
         :param params: Dictionary with the test parameters
         :param env: Dictionary with test environment.
         """
-        self.gagent.get_vcpus()
+        error_context.context("Check cpu info from guest agent.", logging.info)
+        vcpus_info = self.gagent.get_vcpus()
+        cpu_num_qga = len(vcpus_info)
+        for vcpu in vcpus_info:
+            if params.get("os_type") == "linux" and not vcpu["can-offline"]:
+                test.fail("Linux guest cpu can't be offline from qga"
+                          "which isn't expected.")
+            if self.params.get("os_type") == "windows" and vcpu["can-offline"]:
+                test.fail("Windows guest cpu can be offline from qga"
+                          "which isn't expected.")
```

Comment:
@xiagao If some vcpus cann't be offlined, some can be offlined.   As testing will break once hit cann't offline's vcpu. So,Maybe some can-offline vcpus will be skipped.

Reply:
windows guest can't do offline operation,else fail this case.

Reply:
@xiagao linux guest has the same issue.  In addition, This patch cann't cover porlarion case.  For example.  porlation case includes "offline vcpus" and "negative test".

Reply:
@xiangchunfu  you mean linux guest, some vcpu can be offlined,some can be not? As I know, the filed "can-oofline" is fixed by qemu-ga, if not, I think it's qemu-ga product bug.
And the polarion case is covered in function gagent_check_set_vcpus().

Reply:
For function test. Looks good to me.
Acked by: xfu@redhat.com